pkgs/servers/http/nginx/generic.nix

```diff
@@ -158,9 +158,14 @@ stdenv.mkDerivation {
     cp -r ${nginx-doc}/* $doc
   '';
 
-  postInstall = if postInstall != null then postInstall else ''
-    mv $out/sbin $out/bin
-  '';
+  nativeBuildInputs = [ removeReferencesTo ];
+
+  disallowedReferences = map (m: m.src) modules;
+
+  postInstall =
+    let
+      noSourceRefs = lib.concatMapStrings (m: "remove-references-to -t ${m.src} $out/sbin/nginx\n") modules;
```

Comment:
You could not `let` this, but just use the `lib.concatMapStrings […]` directly. That's up to your preference though.

One more and hopefully last real question: Is there any chance that any module (we have in nixpkgs) actually depends on this? As in, needs access to its `.src` store-path?

Reply:
> You could not `let` this, but just use the `lib.concatMapStrings […]` directly. That's up to your preference though.

I like having small, named building blocks. 

> One more and hopefully last real question: Is there any chance that any module (we have in nixpkgs) actually depends on this? As in, needs access to its `.src` store-path?

Sure that's possible. But because of the structure forced on nginx modules (they all have a `config` file in their root folder which fills in some variables and gets interpreted by the main nginx build script, I think it's unlikely. As in "if it works for some modules, it'll probably work for all". I built nginx with a few random modules and the only mention of the of the  `src` store path was indeed in the configure line.

Reply:
We could also try patching it out there then, instead of using `remove-references-to`, which removes it from the the whole binary. Although I'm fine with just going ahead with this and if it does turn out that there is some module that needs a file from its src during runtime, we can still revert and implement that.

Reply:
> I like having small, named building blocks.

And we don't have a style-guide, so neither of us have anything besides "I like this" on their side. You wrote the code, so that's fine by me.